src-tauri/src/blocker.rs

```diff
@@ -82,9 +82,21 @@ impl BitBlocker {
 
 #[cfg(test)]
 mod tests {
+
     use super::*;
     use rand::{seq::SliceRandom, thread_rng, Rng};
 
+    #[quickcheck]
+    fn random_vec_check(bytes: Vec<u8>) -> bool {
+        let blocker = BitBlocker {};
+        let bits = BitVec::from_vec(bytes.clone());
+        if bytes.len() > 0 {
+            bits == blocker.rebuild(blocker.block(bits.clone(), 20, 15), 15)
+        } else {
+            true
+        }
+    }
+
```

Comment:
```
Initial sequence: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
Blocked sequences:
Bl0: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
Bl1: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
thread 'blocker::tests::random_vec_check' panicked at src/blocker.rs:55:13:
Error: No starting block found!
Initial sequence: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
Blocked sequences:
Bl0: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
thread 'blocker::tests::random_vec_check' panicked at src/blocker.rs:55:13:
Error: No starting block found!
Initial sequence: [0, 0, 0, 0, 0, 0, 0, 0]
Blocked sequences:
Bl0: [0, 0, 0, 0, 0, 0, 0, 0]
thread 'blocker::tests::random_vec_check' panicked at src/blocker.rs:41:37:
attempt to subtract with overflow
thread 'blocker::tests::random_vec_check' panicked at /Users/lucyhao/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quickcheck-1.0.3/src/tester.rs:165:28:
[quickcheck] TEST FAILED (runtime error). Arguments: ([0])
Error: "attempt to subtract with overflow"
```

I added a quick check test; can you determine why those errors are happening, esp the subtract with overflow?

Reply:
From just briefly scanning, it seems like `rebuild` should fail for all these sequences (in the sense that they are unrecoverable). However, I will add proper error handling to have the function return an error rather than panic (I'll add error handling with `result` and modify `main.rs` accordingly).

Reply:
What's the update on this?

Reply:
Hi Lucy, I have added in the extensive error handling and patched out the bug I mentioned yesterday. AFAIK this should be ready for merge, but please feel free to pull and check yourself beforehand.

Reply:
Just to clarify though, metadata seems to no longer store per_overlap, and main doesn't seem to implement the block function at all. Is this intended?